### DISCUSSION TITLE
angular build optimizer does not work with monaco editor

### DISCUSSION THREAD

**Discussion by wszgrcy** [microsoft/monaco-editor#3630] [Q&A]
**monaco-editor version:** 0.18.1
**Browser:chrome
**OS:win10

**Steps or JS usage snippet reproducing the issue:**
The problem in this  [issues](https://github.com/angular/angular-cli/issues/14033)
It's not my issues but I have the same with it.
**[error-demourl](https://github.com/petermikitsh/buildoptimizer-error)**
When I run with `ng serve` the monaco has no bug
But When I run with `ng serve --prod`(production mode),the monaco has bug on Console
![bug](https://user-images.githubusercontent.com/1571918/55281321-d83c4280-52ef-11e9-9cd7-42050aad723e.png)

I think the bug belong to `angular-cli` or `monaco` ,but in `angular-cli` ,past few months, it's not fixed.So I hope `monaco` can see it and give a solution.

**Comment by alexdima** [comment]
This looks like it is an issue specific to `@angular-devkit/build-optimizer` --- https://github.com/angular/angular-cli/issues/14033 . We can leave this issue here open for a while, but there is no code change planned on our side to address this...

**Comment by petermikitsh** [comment]
Angular appears to have [no plans](https://github.com/angular/angular-cli/issues/11439#issuecomment-418772149) to change this behavior in the build optimizer. You need to turn off the build optimizer, or have the optimizer [ignore Monaco](https://github.com/angular/angular-cli/issues/14033#issuecomment-507516164).

However, if the error stems from not following best practices (e.g, [object property access should be side-effect free](https://github.com/angular/angular-cli/issues/11439#issuecomment-418766787)), it could be worthwhile to explore changing Monaco so it plays nicely with uglifyjs.  It could benefit the maintainability of Monaco, since developers working on Monaco could unintentionally assume those best practices are valid here.

Both sides present valid arguments. Angular shouldn't impose assumptions on third-party library code, but also making changes to conform to those fairly reasonable assumptions could be overall beneficial to maintainers.

**Comment by alexdima** [comment]
Ignoring the tree shaking optimization for the Monaco Editor sounds reasonable.

The Monaco Editor is shipped from "tree shaken" VS Code sources -- see [here](https://github.com/microsoft/vscode/blob/6ff7bf27c9dc447cd95693debdc5b90910e804b5/build/lib/treeshaking.ts)
